### ISSUE TITLE
Bun.build() hangs on v1.1.6

### ISSUE THREAD

**Issue by dulranga** [oven-sh/bun#10637]
### What version of Bun is running?

1.1.6+e58d67b46

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

Create a script file `build.ts`

add this code

```ts
await Bun.build({
  root: "./src",
  entrypoints: ["./src/main.tsx"],
  outdir: "./dist",
  target: "node",
  minify: true,
  sourcemap: "external",
});
```

run 
```bash
bun run build.ts
```

target: `bun` or `node` both hangs
This worked as aspected in v1.1.4

### What is the expected behavior?

Build the project and make a `./dist` directory

### What do you see instead?

The script doesn't end and it hangs there until interrupts. And no `./dist` folder created

### Additional information

The script worked in bun v1.1.4
The project use [Puppeteer](https://pptr.dev/), Expressjs and MongoDB

**Comment by oz** [comment]
I get a similar error, that could be related to this issue. With bun 1.1.6 (same arch as op), the build hangs when I request a sourcemap.

I mean that this works correctly:

```
bun build --compile --minify index.ts --outfile server
```

But this hangs:

```
bun build --compile --minify --sourcemap index.ts --outfile server
```

(Recommended flags for [production builds](https://bun.sh/docs/bundler/executables#deploying-to-production).)

HTH.

**Comment by dulranga** [comment]
Yes, it's same here. The issue is with   `sourcemap: "external"`
When I remove the sourcemap config, the build works perfectly as normal

**Comment by Jarred-Sumner** [comment]
Duplicate of #10624

**Comment by Jarred-Sumner** [comment]
Fixed in #10625 

This will be included in Bun v1.1.7

to get the fix early, run `bun upgrade --canary`
